src/pipeline.h

```diff
@@ -87,6 +87,9 @@ struct PipelineBaton {
   std::vector<double> flattenBackground;
   bool negate;
   double blurSigma;
+  double brightness;
+  double saturation;
+  double hue;
```

Comment:
I think `hue` can be a `int` (signed, as negative values are acceptable).

Reply:
I actally started out with hue as an integer, but I don't know how to make it work in C type system. `linear` expects vectors with doubles, so it fails with: `error: non-constant-expression cannot be narrowed from type 'int' to 'double' in initializer list [-Wc++11-narrowing]`

Should I somehow convert int back to double in operations.cc?

Reply:
Try something like `{0.0, 0.0, static_cast<double>(hue)}` in the call to `linear()`, which the compiler should be able to use to initialise a `std::vector<double>`.